### PR TITLE
Metrics Webapp

### PR DESCRIPTION
**Pull request by Piercel2022**
This react capstone is about building a mobile web application to check a list of metrics (numeric values) that you will create making use of React and Redux. This webapp is based on the Financial Modeling Prep  and  is the most accurate financial data API. Financial statements  are updated in real time, every statements is audited, standardized, and up to date. 

Testing requirements

    Create unit tests for pure functions in the Redux code.
    Create integration tests for the application using the React Testing Library.
        I mock the access to the API, so that your tests don't send actual requests.
        I mock the connection to the Redux Store.

Technical requirements

    The project is a single-page application (SPA) built with React and Redux.
    The data retrieved from the API should is stored in the Redux store.
     The data that I retrieve are filtered from the API using a Filter stateless component.
    Every page (the main page and the pages for each item)  have a unique route within the SPA.
    The project is deployed and accessible online (there is a link to the online version of the app in the README file).
Stock

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const AppContainer = styled.div`
+  @media (min-width: 768px) {
+    padding: 0 250px;
+  }
+`;
+const App = () => {
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchStocks());
+    // eslint-disable-next-line
```
Comment (AmaduKamara) [inline src/App.js:20]: - [ ] Kindly remove the ESLint disable here. ESLint is there to help you write better code and disabling it defies its purpose. Please fix the linter check on this part 👍

File: src/__tests__/Components.react.test.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        ticker="AAPL"
+        changes={1.2}
+        price="250"
+        changesPercentage="0.23"
+        companyName="Apple Inc."
+      />,
+      { wrapper: MemoryRouter },
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (AmaduKamara) [inline src/__tests__/Components.react.test.js:25]: - [ ] The test on rendering the Card component is failing. Kindly fix that

image

File: src/__tests__/Components.react.test.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        first="2020-02-02"
+        second="Filling Date: 2020-02-05"
+        third="Accepted Date: 2020-02-05"
+      >
+        <span />
+      </DetailsCard>,
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+
```
Comment (AmaduKamara) [inline src/__tests__/Components.react.test.js:68]: - [ ] The test on rendering the Details Card component is failing. Kindly fix that

image

File: src/__tests__/Components.react.test.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    const { container } = render(<Header home />, { wrapper: MemoryRouter });
+
+    const nav = container.querySelector('[data-testid="nav"]');
+    const icons = container.querySelector('[data-testid="icons"]');
+
+    expect(nav.children.length).toBe(2);
+    expect(icons.children.length).toBe(2);
+    expect(await within(nav).findByText('Stock Metrics')).toBeTruthy();
+  });
+});
```
Comment (AmaduKamara) [inline src/__tests__/Components.react.test.js:109]: - [ ] The test on rendering the Header component is failing. Kindly fix that

image

File: src/__tests__/Components.react.test.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+describe('Hero', () => {
+  test('should render correctly', () => {
+    const { asFragment } = render(
+      <Hero
+        text=""
+        stocks={[]}
+      />,
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (AmaduKamara) [inline src/__tests__/Components.react.test.js:120]: The test on rendering the Hero component is failing. Kindly fix that
image

File: src/__tests__/Components.react.test.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+describe('SearchBar', () => {
+  test('should render correctly', () => {
+    const { asFragment } = render(
+      <SearchBar
+        handleSearch={() => {}}
+        data=""
+      />,
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (AmaduKamara) [inline src/__tests__/Components.react.test.js:144]: - [ ] The test on rendering the SearchBar component is failing. Kindly fix that

image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @Piercel2022,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly
✅ No linter error
✅ Good commit messages
✅ PR description is okay
✅ App is beautifully designed and interactive. 👍 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly record a video presenting your project in professional English as required in the project requirements. It is recommended you use Loom to record the video. While recording your presentation, please take note of the points below:
- Record a video presenting the features of the project.
- The maximum length of the video is 5 minutes.
- The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
- The video contains a demo of the project, highlighting all the project features.
- The video contains a highlight about a piece of the code you find interesting or proud of making.

- [ ] It would be nice also for you to add the link of the video presentation to your PR or your README file so the next reviewer can be able to access it. 👍 

- [ ] Kindly give credit to the original creator of the design you used as inspiration to develop this project. 
- The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).

- [ ] Also, you are to deploy the project to either Netlify or Heroku and add the deployed app's link on to your README file. I trust you can do that 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Piercel2022       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] I have made sure to recheck and I confirm that you have not attributed the original creator of the provided design in your README.md file. Kindly recheck the [previous reviewer's](https://github.com/Piercel2022/metrics-webapp/pull/1#pullrequestreview-1103178736) comment he made sure to point this one out to you.

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Piercel2022, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**